src/main/kotlin/no/nav/syfo/aktivitetskrav/database/AktivitetskravVarselRepository.kt

```diff
@@ -47,15 +49,65 @@ class AktivitetskravVarselRepository(private val database: DatabaseInterface) {
             nyttVarsel
         }
 
-    fun getIkkeJournalforte(): List<Triple<PersonIdent, PAktivitetskravVarsel, ByteArray>> = database.getIkkeJournalforteVarsler()
+    fun getIkkeJournalforte(): List<Triple<PersonIdent, PAktivitetskravVarsel, ByteArray>> =
+        database.getIkkeJournalforteVarsler()
 
-    fun getIkkePubliserte(): List<Triple<PersonIdent, PAktivitetskravVarsel, UUID>> = database.getIkkePubliserteVarsler()
+    fun getIkkePubliserte(): List<Triple<PersonIdent, PAktivitetskravVarsel, UUID>> =
+        database.getIkkePubliserteVarsler()
 
     fun updateJournalpostId(varsel: AktivitetskravVarsel, journalpostId: String) =
         database.updateVarselJournalpostId(varsel, journalpostId)
 
     fun setPublished(varsel: AktivitetskravVarsel) =
         database.setPublished(varsel)
+
+    suspend fun getExpiredVarsler(): List<PAktivitetskravVarsel> =
+        withContext(Dispatchers.IO) {
+            val expiredVarsler = database.connection.run {
+                prepareStatement(Queries.selectExpiredVarsler).run {
+                    executeQuery()
+                        .toList { toPAktivitetskravVarsel() }
+                }
+            }
+            expiredVarsler
+        }
+
+    suspend fun updateExpiredVarselPublishedAt(
+        publishedExpiredVarsler: List<PAktivitetskravVarsel>
```

Comment:
Jeg tror det er bedre å oppdatere hvert aktivitetsvarsel for seg etter hver publish til Kafka (da vil de som evnt feiler i cronjob'en bli behandlet ved neste kjøring av cronjob'en).

Dvs ta inn en instans av PAktivitetskravVarsel her i stedet for en liste (og forenkle koden nedenfor).